### PR TITLE
feat(ui,api): display workflow run in UI

### PR DESCRIPTION
**Pull request by sguiheux**
1. Description
1. Related issues
1. About tests
1. Mentions

@ovh/cds

### CODE REVIEW

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#19123.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#19124.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test UI ✔

File: engine/api/v2_queue.go [hunk 1]
```diff
@@ -47,12 +47,14 @@ func (api *API) postJobRunStepHandler() ([]service.RbacChecker, service.Handler)
...
 
-		runjob.StepsContext = stepsContext
+		runjob.StepsStatus = stepsStatus
 		if err := workflow_v2.UpdateJobRun(ctx, tx, runjob); err != nil {
 			return err
 		}
-
-		return sdk.WithStack(tx.Commit())
+		if err := tx.Commit(); err != nil {
+			return err
```
Comment (richardlt) [inline engine/api/v2_queue.go:55]:
```suggestion
			return sdk.WithStack(err)
```

File: engine/cdn/item_logs_handler.go [hunk 2]
```diff
@@ -54,8 +55,14 @@ func (s *Service) getItemLogsStreamHandler() service.Handler {
 
 			// Load last running step
 			var iuID string
-			if filter.JobRunID > 0 {
-				iu, _ := storage.LoadLastItemUnitByJobUnitType(ctx, s.Mapper, s.mustDBWithCtx(ctx), s.Units.LogsBuffer().ID(), filter.JobRunID, sdk.CDNTypeItemStepLog)
+			jobRunInt, err := strconv.Atoi(filter.JobRunID)
+			if err == nil {
+				iu, _ := storage.LoadLastItemUnitByJobUnitType(ctx, s.Mapper, s.mustDBWithCtx(ctx), s.Units.LogsBuffer().ID(), int64(jobRunInt), sdk.CDNTypeItemStepLog)
```
Comment (richardlt) [inline engine/cdn/item_logs_handler.go:60]: No err catch ?

File: engine/cdn/websocket.go [hunk 3]
```diff
@@ -72,14 +73,24 @@ func (s *Service) publishWSEvent(itemUnit sdk.CDNItemUnit) {
...
-	event := sdk.CDNWSEvent{
-		ItemType: itemUnit.Type,
-		JobRunID: apiRefItem.NodeRunJobID,
+	if apiRefItem != nil {
+		event = sdk.CDNWSEvent{
+			ItemType: itemUnit.Type,
+			JobRunID: strconv.FormatInt(apiRefItem.NodeRunJobID, 10),
+		}
+	} else {
+		apiRefLogItem, _ := itemUnit.Item.GetCDNLogApiRefV2()
```
Comment (richardlt) [inline engine/cdn/websocket.go:84]: No err catch ?

**Review by yesnault — APPROVED** [review]

**Review by richardlt — APPROVED** [review]

**Pull request merged**